blas/tpls/KokkosBlas2_syr_tpl_spec_decl_rocblas.hpp

```diff
@@ -0,0 +1,300 @@
+  constexpr int one     = 1;                                                     \
+  const int LDA         = A_is_lr ? A.stride(0) : A.stride(1);                   \
+  rocblas_fill fillMode = (uploChar == 'L' || uploChar == 'l')                   \
+                              ? rocblas_fill_lower                               \
+                              : rocblas_fill_upper;
+
+#define KOKKOSBLAS2_DSYR_ROCBLAS(LAYOUT, EXEC_SPACE, MEM_SPACE,             \
+                                 ETI_SPEC_AVAIL)                            \
+  template <>                                                               \
+  struct SYR<                                                               \
+      EXEC_SPACE,                                                           \
+      Kokkos::View<const double*, LAYOUT,                                   \
+                   Kokkos::Device<EXEC_SPACE, MEM_SPACE>,                   \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,               \
+      Kokkos::View<double**, LAYOUT, Kokkos::Device<EXEC_SPACE, MEM_SPACE>, \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,               \
+      true, ETI_SPEC_AVAIL> {                                               \
+    typedef double SCALAR;                                                  \
+    typedef Kokkos::View<const SCALAR*, LAYOUT,                             \
+                         Kokkos::Device<EXEC_SPACE, MEM_SPACE>,             \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >          \
+        XViewType;                                                          \
+    typedef Kokkos::View<SCALAR**, LAYOUT,                                  \
+                         Kokkos::Device<EXEC_SPACE, MEM_SPACE>,             \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >          \
+        AViewType;                                                          \
+                                                                            \
+    static void syr(const typename AViewType::execution_space& space,       \
+                    const char trans[], const char uplo[],                  \
+                    typename AViewType::const_value_type& alpha,            \
+                    const XViewType& X, const AViewType& A) {               \
+      Kokkos::Profiling::pushRegion("KokkosBlas::syr[TPL_ROCBLAS,double]"); \
+      KOKKOSBLAS2_SYR_ROCBLAS_DETERMINE_ARGS(LAYOUT);                       \
+      if (A_is_ll) {                                                        \
+        KokkosBlas::Impl::RocBlasSingleton& s =                             \
+            KokkosBlas::Impl::RocBlasSingleton::singleton();                \
+        KOKKOS_ROCBLAS_SAFE_CALL_IMPL(                                      \
+            rocblas_set_stream(s.handle, space.hip_stream()));              \
+        KOKKOS_ROCBLAS_SAFE_CALL_IMPL(rocblas_dsyr(                         \
```

Comment:
Let's explicitly `rocblas_set_pointer(rocblas_pointer_mode_host)` before the call, since we're dereferencing a host pointer.

Reply:
Thanks for the hint, Carl. Done.